include/swift/AST/Decl.h

```diff
@@ -819,7 +819,11 @@ class alignas(1 << DeclAlignInBits) Decl {
 
   // Make vanilla new/delete illegal for Decls.
   void *operator new(size_t Bytes) = delete;
+
+  // Work around MSVC error: attempting to reference a deleted function.
```

Comment:
Take a look at this link: https://msdn.microsoft.com/en-us/library/mt723604.aspx
There seems to be some mention of this error, but I'm not sure how applicable this is